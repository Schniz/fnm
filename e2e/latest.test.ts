import { script } from "./shellcode/script.js"
import { Bash, Fish, PowerShell, Zsh } from "./shellcode/shells.js"
import describe from "./describe.js"

for (const shell of [Bash, Zsh, Fish, PowerShell]) {
  describe(shell, () => {
    test(`installs latest`, async () => {
      await script(shell)
        .then(shell.env({}))
        .then(shell.call("fnm", ["install", "--latest"]))
        .then(
          shell.scriptOutputContains(shell.call("fnm", ["ls"]), "latest")
        )
        .then(shell.call("fnm", ["use", "'latest'"]))
        .takeSnapshot(shell)
        .execute(shell)
    })
  })
}
