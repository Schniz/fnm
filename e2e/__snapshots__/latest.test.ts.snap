// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bash installs latest: Bash 1`] = `
"set -e
eval "$(fnm env)"
fnm install --latest
(fnm ls) | grep latest || (echo "Expected output to contain latest" && exit 1)
fnm use 'latest'"
`;

exports[`Fish installs latest: Fish 1`] = `
"fnm env | source
fnm install --latest
begin; fnm ls; end | grep latest; or echo "Expected output to contain latest" && exit 1
fnm use 'latest'"
`;

exports[`PowerShell installs latest: PowerShell 1`] = `
"$ErrorActionPreference = "Stop"
fnm env | Out-String | Invoke-Expression
fnm install --latest
$($__out__ = $(fnm ls | Select-String latest); if ($__out__ -eq $null) { exit 1 } else { $__out__ })
fnm use 'latest'"
`;

exports[`Zsh installs latest: Zsh 1`] = `
"set -e
eval "$(fnm env)"
fnm install --latest
(fnm ls) | grep latest || (echo "Expected output to contain latest" && exit 1)
fnm use 'latest'"
`;
